torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -3021,6 +3108,31 @@ def _finalize_params(fsdp_module: FullyShardedDataParallel) -> None:
                     # reset this flag for cases like "one forward pass + multiple backward passes"
                     self._post_backward_callback_queued = False
 
+        if (self.use_param_exec_order_policy() and self._param_exec_order_prep_stage):
+            self._param_exec_order_prep_stage = False
+            # Let the parameters in self._fsdp_params_exec_order ordered based on
+            # the execution order in the forward pass.
+            self._fsdp_params_exec_order.reverse()
+            for m in self.modules():
+                if m is not self and isinstance(m, FullyShardedDataParallel):
+                    if hasattr(m, "_use_param_exec_order_policy"):
```

Comment:
it's unclear to me why we need to delete these attributes? Also, shall we have some sanity checking, i.e. if one FSDP module has these attributes, the rest should as well?